src/NzbDrone.Core/Organizer/FileNameBuilder.cs

```diff
@@ -283,11 +284,33 @@ public static string TitleThe(string title)
             return title.Trim();
         }
 
-        public static string CleanFileName(string name, bool replace = true)
+        public static string CleanFileName(string name, NamingConfig namingConfig)
         {
+            bool replace = namingConfig.ReplaceIllegalCharacters;
+            string colonReplacementFormat;
+
+            switch ((ColonReplacementFormat) namingConfig.ColonReplacementFormat)

```

Comment:
Might be better suited as a property or method on the enum itself. While we aren't using this anywhere else, we might in the future.